### PR TITLE
Update to Async HTTP Client

### PR DESCRIPTION
**Pull request by cpageler93**

### CODE REVIEW

**Comment by Yasumoto** [comment]
This looks great! Thanks for taking this further  🙏

**Comment by cpageler93** [comment]
@Yasumoto thanks for your feedback. I just improved the search method a bit, added some routes/methods, some commands to the cli tool and updated the readme.

Do you have anything else in mind, that I should add to the project right now? Otherwise I'd like to merge this branch to master and create a new version/release.

File: Sources/JiraSwift/JiraClient.swift [hunk 1]
```diff
@@ -7,112 +7,114 @@
...
+                    else {
+                        throw error
+                    }
+                    throw ClientError.jiraError(jiraResponseError)
+                }
+            }
         }
         
-        public func updateRemoteLinkForIssue(withKey issueKey: String,
+        /*public func updateRemoteLinkForIssue(withKey issueKey: String,
```
Comment (Yasumoto) [inline Sources/JiraSwift/JiraClient.swift]: Delete?

**Review by Yasumoto — APPROVED** [review]
This looks excellent! Definitely think this requires a major version bump for folks too. Thanks again!

**Pull request merged**